### PR TITLE
[15.10] Add support for Strand-specific Coordinate Count datatype

### PR DESCRIPTION
**Pull request by gregvonkuster**

### CODE REVIEW

**Comment by martenson** [comment]
From what I know this is targeting 15.10 because it is necessary for tools that should be deployed on Main this Decemeber (ping @nekrut). As adding a datatype is a fairly isolated change I hope this won't cause any trouble.

xref https://github.com/galaxyproject/galaxy/pull/1164

**Comment by martenson** [comment]
@galaxybot test this

File: lib/galaxy/datatypes/interval.py [hunk 1]
```diff
@@ -1485,6 +1485,83 @@ def sniff( self, filename ):
...
+    MetadataElement(name="columns", default=0, desc="Number of columns", readonly=True, visible=False)
+    MetadataElement(name="column_types", default=[], param=metadata.ColumnTypesParameter, desc="Column types", readonly=True, visible=False, no_value=[])
+
+    def __init__(self, **kwd):
+        """
+        Initialize scidx datatype.
+        """
+        Tabular.__init__(self, **kwd)
+        # Don't set column names since the first
+        # line of the dataset displays them.
```
Comment (bgruening) [inline lib/galaxy/datatypes/interval.py:1510]: I think it's a nice feature to include the column names as well. You could see them even if you have scrolled to the bottom.

File: lib/galaxy/datatypes/interval.py [hunk 2]
```diff
@@ -1485,6 +1485,83 @@ def sniff( self, filename ):
         return False
 
 
+class ScIdx(Tabular):
+    """
+    Support for the Strand-specific coordinate count datatype which is used
```
Comment (bgruening) [inline lib/galaxy/datatypes/interval.py]: @gregvonkuster is this needed here?

File: lib/galaxy/datatypes/interval.py [hunk 3]
```diff
@@ -1485,6 +1485,83 @@ def sniff( self, filename ):
...
+    is used by the Center for Eukaryotic Gene Regulation labs at Penn State
+    University.
+
+    ScIdx files are 1-based and consist of strand-specific coordinate counts.
+    They always have 5 columns, and the first row is the column labels:
+    'chrom', 'index', 'forward', 'reverse', 'value'.
+    Each line following the first consists of data:
+    chromosome name (type str), peak index (type int), Forward strand peak
+    count (type int), Reverse strand peak count (type int) and value (type int).
+    The value of the 5th 'value' column is the sum of the forward and end
```
Comment (bgruening) [inline lib/galaxy/datatypes/interval.py]: and end?

File: lib/galaxy/datatypes/interval.py (hunk 1 above)
Comment (gregvonkuster) [inline lib/galaxy/datatypes/interval.py:1510]: @bgruening Updates are here: https://github.com/gregvonkuster/galaxy/commit/21fad8ac88a46996b2ff2719527709374f0c9913  Thanks!

**Comment by martenson** [comment]
thanks @gregvonkuster !!

**Comment by gregvonkuster** [comment]
Thanks for merging.

**Pull request merged**